libraries/ms-common/inc/can_msg.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+// Defines the CAN message type
+// This is kept in a separate file to prevent cyclic dependencies
+#include <stdint.h>
+#include <stddef.h>
+
+#define CAN_MSG_INVALID_ID (UINT16_MAX)
+#define CAN_MSG_INVALID_DEVICE (UINT16_MAX)
+
+#define CAN_MSG_MAX_DEVICES (1 << 4)
+#define CAN_MSG_MAX_IDS (1 << 6)
+
+// TODO(ELEC-55): determine which messages are considered "critical"
+#define CAN_MSG_IS_CRITICAL(msg) ((msg)->msg_id < 14)
+
+#define CAN_MSG_SET_RAW_ID(can_msg, can_id) \
+do { \
+  CANId id = { .raw = (can_id) }; \
+  (can_msg)->source_id = id.source_id; \
+  (can_msg)->msg_id = id.msg_id; \
+  (can_msg)->type = id.type; \
+} while (0)
+
+typedef enum {
+  CAN_MSG_TYPE_DATA = 0,
+  CAN_MSG_TYPE_ACK,
+  CAN_MSG_NUM_TYPES
```

Comment:
Technically we have some that don't follow that convention (`*_mcu`), but I'll change this one.